File: catalog-service/src/main/kotlin/it/polito/ecommerce/catalogservice/dto/kafkadtos/UserCreatedDTO.kt
package it.polito.ecommerce.catalogservice.dto.kafkadtos

import it.polito.ecommerce.catalogservice.domain.Rolename
import java.time.LocalDateTime

data class UserCreatedDTO(
    val id: Long,
    val username: String,
    val email: String,
    val roles: Set<Rolename>,
    val name: String,
    val surname: String,
    val deliveryAddress: String,
    val emailVerificationTokenInfo: UserCreatedEmailVerificationTokenInfoDTO
)

data class UserCreatedEmailVerificationTokenInfoDTO(
    val expirationDate: LocalDateTime,
    val token: String,
)

File: mail-service/src/dtos/UserCreatedDTO.ts
export interface UserCreatedEmailVerificationTokenInfoDTO {
  expirationDate: Date;
  token: string;
}

export interface UserCreatedDTO {
  id: number;
  username: string;
  email: string;
  roles: string[];
  emailVerificationTokenInfo: UserCreatedEmailVerificationTokenInfoDTO;
  name: string;
  surname: string;
  deliveryAddress: string;
}

